### PR TITLE
📘DOC: Fix URL normalization for the Left Sidebar in docs

### PR DESCRIPTION
**Pull request by jasikpark**
## Changes

Fixes the highlighting of the current page in `LeftSidebar.astro`

### CODE REVIEW

File: docs/src/components/LeftSidebar/LeftSidebar.astro [hunk 1]
```diff
@@ -2,7 +2,9 @@
 import { SIDEBAR } from '../../config.ts';
 import { getLanguageFromURL } from '../../util.ts';
 const {currentPage} = Astro.props;
-const currentPageMatch = currentPage.slice(1);
+
+// Get the slug w/o a leading or trailing slash
+const currentPageMatch = currentPage.slice(1, (currentPage[currentPage.length-1] === '/') ? -1 : currentPage.length);
```
Comment (FredKSchott) [inline docs/src/components/LeftSidebar/LeftSidebar.astro:7]: we actually have a util helper for this!

```
import {addTrailingSlash} from '../../../util.ts';
```

If you're trying to do the opposite, you can add these helpers to `util.ts` and then use those, as well:

```

/** Remove \ and / from beginning of string */
export function removeLeadingSlash(path: string) {
  return path.replace(/^[/\\]+/, '');
}

/** Remove \ and / from end of string */
export function removeTrailingSlash(path: string) {
  return path.replace(/[/\\]+$/, '');
}
```

Comment (jasikpark) [reply docs/src/components/LeftSidebar/LeftSidebar.astro:7]: Where is that imported from? I like the idea of the docs site being self-contained

Comment (jasikpark) [reply docs/src/components/LeftSidebar/LeftSidebar.astro:7]: nvm, found it! https://github.com/snowpackjs/astro/blob/main/docs/src/util.ts

Comment (jasikpark) [reply docs/src/components/LeftSidebar/LeftSidebar.astro:7]: Feel free to make a change like that, I probably won't touch this till tomorrow night.

**Comment by okikio** [comment]
@FredKSchott I added the remove trailing and leading slash functions, but I get this error while trying out the `docs/` site, I think it comes from the `@docsearch/react` (I created a fork with the trailing and leading slash function, https://github.com/okikio/astro)
```bash
Error: Minified React error #321; visit https://reactjs.org/docs/error-decoder.html?invariant=321 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at S (/_snowpack/pkg/react.v17.0.2/common/index-29632863.js:29:330)
    at Object.exports.useState (/_snowpack/pkg/react.v17.0.2/common/index-29632863.js:34:277)
    at Object.Search (/_astro/src/components/Header/Search.js:12:41)
    at x (/_snowpack/pkg/preact-render-to-string.v5.1.19.js:5:2375)
    at Object.m [as default] (/_snowpack/pkg/preact-render-to-string.v5.1.19.js:5:1063)
    at renderToStaticMarkup (/_snowpack/link/packages/renderers/renderer-preact/server.js:27:26)
    at Object.check (/_snowpack/link/packages/renderers/renderer-preact/server.js:14:20)
    at resolveRenderer (/_snowpack/link/packages/astro/dist/internal/__astro_component.js:82:40)
    at __astro_component_internal (/_snowpack/link/packages/astro/dist/internal/__astro_component.js:172:26)
    at Object.h (/_snowpack/link/packages/astro/dist/internal/h.js:48:12)
    at async Promise.all (index 0)
    at async Object.h (/_snowpack/link/packages/astro/dist/internal/h.js:46:20)
    at async Promise.all (index 4)
    at async Object.h (/_snowpack/link/packages/astro/dist/internal/h.js:46:20)
    at async Promise.all (index 1)
    at async Object.h (/_snowpack/link/packages/astro/dist/internal/h.js:46:20)
```

**Comment by FredKSchott** [comment]
That's odd! I'd be very suprised if that is related to adding that small helper function to the repo, though

**Comment by okikio** [comment]
> That's odd! I'd be very suprised if that is related to adding that small helper function to the repo, though

@FredKSchott The error relates to react specifically, I've already tested out the `docs/` with and without the helper functions, and they don't affect anything.

**Comment by jasikpark** [comment]
Fixes https://github.com/snowpackjs/astro/issues/1307

**Pull request merged**